### ISSUE TITLE
Getting exception: System.IndexOUtOfRangeException: 'Index was outside the bounds of the array.'

### ISSUE THREAD

**Issue by jlinkSC** [JoshClose/CsvHelper#1449]
Running into the above exception when trying to read a csv with 135 columns. All were setup as string.

Any help you can provide?

Here's the code I'm using:

```
const string csvFilePath = @"path\to\file\with\135\colums";
using (var dt = new DataTable())
{
    using (var reader = new StreamReader(csvFilePath))
    using (var csv = new CsvReader(reader,CultureInfo.InvariantCulture))
    {
        using (var dr = new CsvDataReader(csv))
        {
            dt.Load(dr);
        }
    }
}
```

**Comment by JoshClose** [comment]
I'll need some more information. This works for me.

```cs
void Main()
{
	using (var stream = new MemoryStream())
	using (var reader = new StreamReader(stream))
	using (var writer = new StreamWriter(stream))
	using (var csvReader = new CsvReader(reader, CultureInfo.InvariantCulture))
	using (var csvWriter = new CsvWriter(writer, CultureInfo.InvariantCulture))
	{
		var columns = 150;
		
		for (var i = 1; i <= columns; i++)
		{
			csvWriter.WriteField($"Header{i}");
		}
		csvWriter.NextRecord();
		
		for (var i = 1; i <= columns; i++)
		{
			csvWriter.WriteField(i);
		}
		csvWriter.NextRecord();
		
		writer.Flush();
		stream.Position = 0;
		
		using (var dr = new CsvDataReader(csvReader))
		{
			var dt = new DataTable();
			dt.Load(dr);	
			
			dt.Dump();
		}
	}
}
```

**Comment by mudnug** [comment]
I also tested the issue code with a physical file and the .NET CORE CsvHelper nuget and don't get any exception. If there were an issue, it could be helpful to have the full exception stack-trace.

using CsvHelper;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Data;
using System.Globalization;
using System.IO;

namespace SampleCsvHelperCORE
{
    [TestClass]
    public class UnitTest1
    {
        [TestMethod]
        public void TestMethod1()
        {
            const string csvFilePath = @"random-data.csv";
            using (var dt = new DataTable())
            {
                using (var reader = new StreamReader(csvFilePath))
                using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
                {
                    using (var dr = new CsvDataReader(csv))
                    {
                        dt.Load(dr);
                    }
                }
            }
        }
    }
}

**Comment by nrspring** [comment]
I have a similar issue.  The source document has a trailing comma so there appears to be one more column than the header.  I added the new column in the header and it worked fine.

**Comment by mudnug** [comment]
Using your description I was able to reproduce the issue

    System.IndexOutOfRangeException: Index was outside the bounds of the array.
  Stack Trace: 
    CsvDataReader.GetValues(Object[] values)
    CommonLanguageSubsetDataReader.GetValues(Object[] values)
    SchemaMapping.LoadDataRow()
    DataAdapter.FillLoadDataRow(SchemaMapping mapping)
    DataAdapter.FillFromReader(DataSet dataset, DataTable datatable, String srcTable, DataReaderContainer dataReader, Int32 startRecord, Int32 maxRecords, DataColumn parentChapterColumn, Object parentChapterValue)
    DataAdapter.Fill(DataTable[] dataTables, IDataReader dataReader, Int32 startRecord, Int32 maxRecords)
    DataTable.Load(IDataReader reader, LoadOption loadOption, FillErrorEventHandler errorHandler)
    DataTable.Load(IDataReader reader)

**Comment by mudnug** [comment]
@JoshClose Should the last row extra column/comma be ignored in the case?

**Comment by nrspring** [comment]
In my case I ended up with re-writing the header in pre-processing to add another column header.  It would be nice (and maybe it's there and I couldnt find it) to tell it to ignore any extra columns and not blow up.

**Comment by JoshClose** [comment]
Thanks @mudnug and @nrspring. I'm able to reproduce it now.

**Comment by JoshClose** [comment]
Fixed in version 15.0.2. Will be on NuGet shortly. https://www.nuget.org/packages/CsvHelper/15.0.2
